File: java/src/main/java/it/giannibombelli/wsc2026/payment/application/commands/RefundTransaction.java
package it.giannibombelli.wsc2026.payment.application.commands;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.application.Command;
import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.payment.domain.payment.PaymentId;

public record RefundTransaction(PaymentId aggregateId, Money amount)
    implements Command<PaymentId> {
    public RefundTransaction {
        Require.requireArgument(aggregateId, "aggregateId");
        Require.requireArgument(amount, "amount");
    }
}

File: typescript/src/payment/application/commands/refundTransaction.ts
// Command per richiedere un rimborso.

import type { Command } from '@/common/application/command.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { PaymentId } from '../../domain/payment/paymentId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type RefundTransaction = Command<'RefundTransaction', PaymentId> & {
  readonly amount: Money;
};

export function refundTransaction(aggregateId: PaymentId, amount: Money): RefundTransaction {
  requireArgument(aggregateId, 'aggregateId');
  requireArgument(amount, 'amount');
  return { kind: 'RefundTransaction', aggregateId, amount };
}
